core/src/main/scala/org/ensime/indexer/domain.scala

```diff
@@ -47,27 +47,34 @@ case class ClassName(pack: PackageName, name: String) extends FullyQualifiedName
     if (pack.path.isEmpty) name
     else ClassName.cleanupPackage(pack.fqnString + "." + name)
 
-  def internalString =
-    "L" + (if (pack.path.isEmpty) name else pack.path.mkString("/") + "/" + name) + ";"
+  def internalString = name match {
```

Comment:
It's a derivative value, though - with a strict definition in terms of pack and name. I don't think it should be open to any kind of string.